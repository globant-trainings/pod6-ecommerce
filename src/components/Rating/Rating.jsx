import React from 'react';
import PropTypes from 'prop-types';
import './rating.css';

function getClassName(index, rating) {
  if(index <= rating) return "star on"
  else if(index - 0.5 <= rating) return "star half"
  return "star"
}

const Rating = ({
  rating
}) => {
  return (
    <div data-testid="star-rating">
      {[...Array(5)].map((star, index) => {
        index += 1;
        return (
          <span
            key={index}
            className={getClassName(index, rating)}
          >
          </span>
        );
      })}
    </div>
  );
}

Rating.propTypes = {
  rating: PropTypes.number,
}


export default Rating;
