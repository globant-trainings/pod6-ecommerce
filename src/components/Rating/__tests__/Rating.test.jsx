import Rating from '../Rating';
import { render, screen } from '@testing-library/react';

describe('<Rating/>', () => {
  test('Should render three stars yellow follwed by two blanks', () => {
    render(<Rating rating={3} />);

    let stars = screen.getByTestId("star-rating");

    expect(stars).toMatchSnapshot('Check rendered yellow stars to be the first three');
  });
  test('Should render two and a half stars yellow follwed by two and a half blanks', () => {
    render(<Rating rating={2.5} />);

    let stars = screen.getByTestId("star-rating");

    expect(stars).toMatchSnapshot('Check rendered yellow stars to be the first two and a half');
  });
  test('Should render two stars yellow follwed by three blanks', () => {
    render(<Rating rating={2.49} />);

    let stars = screen.getByTestId("star-rating");

    expect(stars).toMatchSnapshot('Check rendered yellow stars to be the first two');
  });

});