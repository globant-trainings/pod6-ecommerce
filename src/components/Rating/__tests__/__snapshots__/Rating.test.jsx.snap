// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rating/> Should render three stars yellow follwed by two blanks: Check rendered yellow stars to be the first three 1`] = `
<div
  data-testid="star-rating"
>
  <span
    class="star on"
  />
  <span
    class="star on"
  />
  <span
    class="star on"
  />
  <span
    class="star"
  />
  <span
    class="star"
  />
</div>
`;

exports[`<Rating/> Should render two and a half stars yellow follwed by two and a half blanks: Check rendered yellow stars to be the first two and a half 1`] = `
<div
  data-testid="star-rating"
>
  <span
    class="star on"
  />
  <span
    class="star on"
  />
  <span
    class="star half"
  />
  <span
    class="star"
  />
  <span
    class="star"
  />
</div>
`;

exports[`<Rating/> Should render two stars yellow follwed by three blanks: Check rendered yellow stars to be the first two 1`] = `
<div
  data-testid="star-rating"
>
  <span
    class="star on"
  />
  <span
    class="star on"
  />
  <span
    class="star"
  />
  <span
    class="star"
  />
  <span
    class="star"
  />
</div>
`;
